public/scripts/world-info.js

```diff
@@ -1898,6 +1898,44 @@ function getWIElement(name) {
     return wiElement;
 }
 
+/**
+ * Adds missing fields to WI entries that are present in the entry template, but not in the data.
+ * Additionally verify that array/object fields are of the expected type.
+ * @param {any[]} data WI entries
+ * @returns {any[]} Data with backfilled fields
+ */
+function addMissingWorldInfoFields(data) {
+    data.forEach((entry) => {
+        // Add missing fields from the template
+        Object.entries(newWorldInfoEntryTemplate).forEach(([key, value]) => {
+            if (!Object.hasOwn(entry, key)) {
+                entry[key] = structuredClone(value);
+            }
+        });
+
+        // Ensure that the key is always an array
+        if (!Array.isArray(entry.key)) {
+            entry.key = [];
+        }
+
+        // Ensure that the keysecondary is always an array
+        if (!Array.isArray(entry.keysecondary)) {
+            entry.keysecondary = [];
+        }
+
+        // Ensure that the characterFilter is an object with the expected structure
+        if (!entry.characterFilter || typeof entry.characterFilter !== 'object') {
```

Comment:
Funnily enough I wanted to do a very quick mockup test to see this works too, and replaced my `characterFilter` with an empty array.
Realizing later that `[]` is of course treaded as object type.
So this might not catch all misformatted data (we would have to check for each sub field of the object for that), but well.